cmd/ci-operator/main.go

```diff
@@ -2176,13 +2184,46 @@ func (o *options) getInjectTest() (*api.MetadataWithTest, error) {
 	return api.MetadataTestFromString(o.injectTest)
 }
 
-// loadConfig loads the standard configuration path, env, or configresolver (in that order of priority)
-func (o *options) loadConfig(info *api.Metadata) (*api.ReleaseBuildConfiguration, error) {
+type gcsFileReader interface {
+	Read(filePath string) ([]byte, error)
+}
+
+type bucketReader struct {
+	opener prowio.Opener
+}
+
+func (b bucketReader) Read(filePath string) ([]byte, error) {
+	content, err := prowio.ReadContent(context.Background(), logrus.WithField("file", filePath), b.opener, filePath)
+	return content, err
+}
+
+const (
+	configSpecVar       = "CONFIG_SPEC"
+	configSpecGcsUrlVar = "CONFIG_SPEC_GCS_URL"
+	unresolvedConfigVar = "UNRESOLVED_CONFIG"
+	gcsBucket           = "test-platform-results"
+)
+
+// loadConfig loads the standard configuration path, env, gcs bucket env, or configresolver (in that order of priority)
+func (o *options) loadConfig(info *api.Metadata, gcsReader gcsFileReader) (*api.ReleaseBuildConfiguration, error) {
 	var raw string
 
-	configSpecEnv, configSpecSet := os.LookupEnv("CONFIG_SPEC")
-	unresolvedConfigEnv, unresolvedConfigSet := os.LookupEnv("UNRESOLVED_CONFIG")
+	configSpecEnv, configSpecSet := os.LookupEnv(configSpecVar)
+	configSpecGCSEnv, configSpecGCSSet := os.LookupEnv(configSpecGcsUrlVar)
+	unresolvedConfigEnv, unresolvedConfigSet := os.LookupEnv(unresolvedConfigVar)
 
+	decodeAndUnzip := func(rawConfig string) (string, error) {
+		// if being run by pj-rehearse, config spec may be base64 and gzipped
+		if decoded, err := base64.StdEncoding.DecodeString(rawConfig); err != nil {
+			return rawConfig, nil
```

Comment:
Why don't we propagate `err` here?

Reply:
I was just maintaining the original behavior. I believe that the intention is to assume the rawConfig is not encoded or zipped in that case. I guess that logging or failing would not be useful.